### PR TITLE
Inrepoconfig: Add validation

### PR DESCRIPTION
**Pull request by alvaroaleman**
Towards #13370, with this PR inrepoconfig can be validated via the checkconfig tool and with that, the feature becomes usable, albeit being completely undocumented so far.

/assign @stevekuznetsov

### CODE REVIEW

File: prow/cmd/checkconfig/main.go [hunk 1]
```diff
@@ -125,11 +128,20 @@ func getAllWarnings() []string {
...
 	if o.configPath == "" {
 		return errors.New("required flag --config-path was unset")
 	}
+
+	if o.prowYAMLPath != "" && o.prowYAMLRepoName == "" {
+		return errors.New("--prow-yaml-repo-path requires --prow-yaml-repo-name to be set")
+	}
+	if o.prowYAMLRepoName != "" {
+		if o.prowYAMLPath == "" {
+			o.prowYAMLPath = fmt.Sprintf("/home/prow/go/src/github.com/%s/.prow.yaml", o.prowYAMLRepoName)
```
Comment (stevekuznetsov) [inline prow/cmd/checkconfig/main.go]: How common is this sort of path?

File: prow/plugins/trigger/generic-comment.go [hunk 2]
```diff
@@ -55,7 +55,10 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 
 	presubmits, err := c.Config.GetPresubmits(c.GitClient, org+"/"+repo, refGetter.BaseSHA, refGetter.HeadSHA)
 	if err != nil {
-		return fmt.Errorf("failed to get presubmits: %v", err)
```
Comment (stevekuznetsov) [inline prow/plugins/trigger/generic-comment.go]: This seems pretty large in functionality change -- is it reasonable to test this flow?

File: prow/cmd/checkconfig/main.go (hunk 1 above)
Comment (alvaroaleman) [reply prow/cmd/checkconfig/main.go]: This is what you get from the decoration utils, so rather common I would say

Comment (alvaroaleman) [reply prow/cmd/checkconfig/main.go]: https://github.com/kubernetes/test-infra/blob/0d180418583d65518a976b339219861ac5fafb5b/prow/pod-utilities.md

File: prow/plugins/trigger/generic-comment.go (hunk 2 above)
Comment (alvaroaleman) [reply prow/plugins/trigger/generic-comment.go]: You are right in that its not completely intuitive at first why its needed and its super important, so I have refactored the code a bit and then added tests

**Pull request merged**